lib/heuristics/concerns/scheduling_data_initialisation.rb

```diff
@@ -192,8 +190,18 @@ def adapt_services_data(vrp)
   end
 
   def collect_indices(vrp)
-    @services_data.collect{ |_id, data| data[:points_ids] }.flatten.uniq.sort.each{ |point_id|
-      @indices[point_id] = vrp.points.find{ |pt| pt.id == point_id }&.matrix_index
+    vrp.vehicles.each{ |vehicle|
+      @indices[vehicle.start_point.id] = vehicle.start_point.matrix_index if vehicle.start_point
+      @indices[vehicle.end_point.id] = vehicle.end_point.matrix_index if vehicle.end_point
+    }
+
+    vrp.services.each{ |service|
+      [service.activity ? [service.activity] : service.activities].flatten.each{ |activity|
+        @indices[activity.point.id] = activity.point.matrix_index
```

Comment:
Isn't this formulation possible ?
(service.activity ? [service.activity] : service.activities).each

Reply:
you are right, that would be better